### PR TITLE
BUG 1913469: LOG-453 Expose recommendations based on cluster health

### PR DESCRIPTION
**Pull request by huikang**
### Description
- Provide links in the alert message
- Descriptions about the alerts in the troubleshooting.md

/cc @lukas-vlcek 
/assign @ewolinetz 

### Links
- Github issue: https://issues.redhat.com/browse/LOG-453
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1913469

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -8,4 +8,9 @@ Infrastructure logs are visible from the `Global` tenant and require `administra
...
 ```
 oc adm policy add-cluster-role-to-user cluster-admin kube:admin
-```
\ No newline at end of file
+```
+
+## Elasticsearch
+
+### Why do I receive the alert `ElasticsearchHighFileDescriptorUsage`
+The cause of this alert is due to [insufficient file descriptors on the node] (https://www.elastic.co/guide/en/elasticsearch/reference/current/file-descriptors.html).
```
Comment (periklis) [inline docs/troubleshooting.md:16]: I suggest to add a new markdown file named `alerts.md` and structure it like this:
https://gitlab.cee.redhat.com/observatorium/configuration/-/blob/master/docs/sop/observatorium.md

This way you can link each alert to its subsection in `alerts.md`. WDYT?

Comment (huikang) [reply docs/troubleshooting.md:16]: Agreed. Thanks.

File: docs/alerts.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - [ElasticsearchClusterNotHealthyYellow](#ElasticsearchClusterNotHealthyYellow)
+    - [ElasticsearchWriteRequestsRejectionJumps](#ElasticsearchWriteRequestsRejectionJumps)
+    - [ElasticsearchNodeDiskWatermarkDiskLowReached](#ElasticsearchNodeDiskWatermarkDiskLowReached)
+    - [ElasticsearchNodeDiskWatermarkDiskHighReached](#ElasticsearchNodeDiskWatermarkDiskHighReached)
+    - [ElasticsearchNodeDiskWatermarkFloodReached](#ElasticsearchNodeDiskWatermarkFloodReached)
+    - [ElasticsearchJVMHeapUseHigh](#ElasticsearchJVMHeapUseHigh)
+    - [AggregatedLoggingSystemCPUHigh](#AggregatedLoggingSystemCPUHigh)
+    - [ElasticsearchProcessCPUHigh](#ElasticsearchProcessCPUHigh)
+    - [ElasticsearchDiskSpaceRunningLow](#ElasticsearchDiskSpaceRunningLow)
+    - [ElasticsearchHighFileDescriptorUsage](#ElasticsearchHighFileDescriptorUsage)
```
Comment (ewolinetz) [inline docs/alerts.md:16]: Given we link to these directly from our alerts, can we make the titles more human readable?
e.g. "Elasticsearch Cluster Health is Red", "Elasticsearch Cluster Health is Yellow", etc

File: docs/alerts.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+<!-- /TOC -->
+
+## ElasticsearchClusterNotHealthyRed
+
+Cluster health status is RED. At least one primary shard and its replicas are not allocated to a node.
```
Comment (ewolinetz) [inline docs/alerts.md:22]: Can we rephrase this slightly and also add some preliminary "things to check for" (even if its sparse initially)

```
## Elasticsearch Cluster Health is Red

At least one primary shard and its replicas are not yet allocated to a node.

### Troubleshooting
```

File: docs/alerts.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## ElasticsearchClusterNotHealthyYellow
+
+Cluster health status is Yellow. Replica shards for at least one index aren't allocated to nodes.
```
Comment (ewolinetz) [inline docs/alerts.md:26]: s/index aren't/primary shard isn't/

File: docs/alerts.md (hunk 3 above)
Comment (ewolinetz) [reply docs/alerts.md:22]: This could apply to the other sections as well

Comment (huikang) [reply docs/alerts.md]: Updated all section titles with the troubleshooting subsection. Please review the change. Thanks.

File: docs/troubleshooting.md [hunk 5]
```diff
@@ -8,4 +8,4 @@ Infrastructure logs are visible from the `Global` tenant and require `administra
 `kube:admin` by default does not have the correct permissions to be given the admin role.   See the [access control](access-control.md) documentation for additional informations.  You may grant the permissions by:
 ```
 oc adm policy add-cluster-role-to-user cluster-admin kube:admin
-```
\ No newline at end of file
+```
```
Comment (ewolinetz) [inline docs/troubleshooting.md:11]: not sure what changed here... can we do a `git checkout upstream/master -- docs/troubleshooting.md` just to clear this file from the list of changed ones?

File: docs/alerts.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+At least one primary shard and its replicas are not allocated to a node.
+
+### Troubleshooting
+
+Cluster nodes could be failed or the Elasticsearch process crashes due to heavy load.
+
+## Elasticsearch Cluster Healthy is Yellow
+
+Replica shards for at least one primary shard aren't allocated to nodes.
```
Comment (ewolinetz) [inline docs/alerts.md:30]: s/aren't/isn't/

Comment (huikang) [reply docs/alerts.md]: Fixed

File: docs/troubleshooting.md (hunk 5 above)
Comment (huikang) [reply docs/troubleshooting.md]: It is cleared. Thanks for your instruction.

File: docs/alerts.md (hunk 6 above)
Comment (ewolinetz) [reply docs/alerts.md]: i clearly didn't read the whole sentence before i made this comment... you were correct before with "aren't"

Comment (huikang) [reply docs/alerts.md:30]: Np. Updated

**Pull request merged**